src/rates/merger/index.ts

```diff
@@ -0,0 +1,366 @@
+import { calculatePercentageDifference } from 'src/shared/utils';
+import { SourceTickers, TickerPrice, Tickers } from '../api/dto/tickers.dto';
+import * as strategies from './strategy';
+import { Notifier } from 'src/global/notifier/notifier.service';
+
+export type StrategyName = keyof typeof strategies;
+
+interface ResourceWeights {
+  [resourceName: string]: number;
+}
+
+interface RatesMergerOptions {
+  baseCoins: string[];
+  decimals: number;
+  weights: ResourceWeights;
+  priorities: string[];
+  threshold: number;
+  groupPercentage: number;
+  minSources: number;
+  rateLifetime: number;
+}
+
+export interface SourcePrice {
+  source: string;
+  price: number;
+  priority: number;
+  weight: number;
+}
+
+interface PriceGroup {
+  weight: number;
+  prices: SourcePrice[];
+}
+
+export abstract class RatesMerger {
+  tickers: Tickers;
+
+  sourceTickers: SourceTickers;
+  rateDifferences: [string, string][] = [];
```

Comment:
```suggestion
  rateDifferences: Array<[string, string]> = [];
```

Suggestion: Much cleaner in my opinion. Resolve if you leave as is.